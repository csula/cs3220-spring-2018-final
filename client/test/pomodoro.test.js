/*eslint-env node, jest */

import PomodoroComponent from '../src/views/pomodoro';
require('@skatejs/ssr/register');
const render = require('@skatejs/ssr');

describe('Pomodoro Timer Component', () => {
	beforeEach(() => {
		window.customElements.define('pomodoro-timer', PomodoroComponent);
	});

	it('should be able to render default pomodoro timer to be 15 seconds', done => {
		render(new PomodoroComponent()).then(html => {
			expect(html).toMatchSnapshot();
			done();
		});
	});

	it('should be able to render default pomodoro timer to be 15 minutes', done => {
		const pomodoroTimer = new PomodoroComponent();
		pomodoroTimer.countdown = 15 * 60;
		render(pomodoroTimer).then(html => {
			expect(html).toMatchSnapshot();
			done();
		});
	});
});
