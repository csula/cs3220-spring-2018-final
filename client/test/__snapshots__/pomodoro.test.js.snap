// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pomodoro Timer Component should be able to render default pomodoro timer to be 15 minutes 1`] = `

<pomodoro-timer>
  <div class="timer">
    Timer: 15:00
  </div>
</pomodoro-timer>

`;

exports[`Pomodoro Timer Component should be able to render default pomodoro timer to be 15 seconds 1`] = `

<pomodoro-timer>
  <div class="timer">
    Timer: 00:15
  </div>
</pomodoro-timer>

`;
